### PR TITLE
Ruby 3

### LINKED ISSUE #70: Update Ruby

### PR DESCRIPTION
**Pull request by whereismyjetpack**
Fixes #70 

updates ruby
updates niftany
applies all new cops sans `Style/MapToHash:` which broke a tika spec.

### CODE REVIEW

**Comment by whereismyjetpack** [comment]
Deployed this code and verified that the main functions still Fuction

```
│ I, [2022-08-01T14:09:10.020106 #9]  INFO -- : No virus found for /tmp/20220801-9-8shcyg                        │
│ I, [2022-08-01T14:09:12.991518 #9]  INFO -- : Text extracted successfully                                      │
│ [ActiveJob] [MetadataListener::Job] [1e293b15-9c74-4db6-8a22-a14b57108ff0] Performed MetadataListener::Job (Jo │
│ b ID: 1e293b15-9c74-4db6-8a22-a14b57108ff0) from Async(metadata) in 30434.43ms                                 │
│ 2022-08-01T18:09:12.992Z pid=9 tid=41d class=MetadataListener::Job jid=536f15d1816a978293b96f4c elapsed=30.451 │
│  INFO: done
```

**Review by ajkiessl — APPROVED** [review]
If everything is working, then this looks good to go.  I just had one question, but nothing needs changed.

File: lib/metadata_listener/job.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ def perform(path:, endpoint: nil, api_token: nil, services: [])
       services << :virus if services.empty?
 
       services.map do |key|
-        report(key).call(path: file.body.path, endpoint: endpoint, api_token: api_token)
```
[the remark this answers is not in the document]
Comment (ajkiessl) [reply lib/metadata_listener/job.rb:18]: Sorry, meant to tag the line below this.

[the remark this answers is not in the document]
Comment (ajkiessl) [reply lib/metadata_listener/job.rb:18]: https://github.com/psu-libraries/metadata-listener/pull/74/files#diff-f994c4d28269c4906e0c0e73e457aba84cf4035f79bb13c57f55f2b47891eaa3R18

[the remark this answers is not in the document]
Comment (whereismyjetpack) [reply lib/metadata_listener/job.rb:18]: it's a ruby3 / rubocop thing. "keyword coercion"

https://medium.com/building-rigup/hash-and-keyword-coercion-213425f69719

**Pull request merged**